src/ui/ui_get_public_key.c

```diff
@@ -133,9 +140,38 @@ UX_STEP_VALID(ux_approve_pk_flow_3_step, pb,
 UX_DEF(ux_approve_pk_flow, &ux_approve_pk_flow_1_step,
        &ux_approve_pk_flow_2_step, &ux_approve_pk_flow_3_step);
 
+#elif defined(HAVE_NBGL)
+
+
+static void callback_match(bool match) {
+    if (match) {
+        io_exchange_with_code(EXCEPTION_OK, 32);
+    } else {
+        io_exchange_with_code(EXCEPTION_USER_REJECTED, 0);
+    }
+    ui_idle();
+}
+
+static void callback_export(bool accept) {
+    if (accept) {
+        nbgl_useCaseAddressConfirmation((const char *) gpk_ctx.full_key, callback_match);
+    } else {
+        io_exchange_with_code(EXCEPTION_USER_REJECTED, 0);
+        ui_idle();
+    }
+}
+
+static void ui_get_public_key_nbgl(void) {
+    nbgl_useCaseChoice(&C_icon_hedera_64x64, "Export Public Key?", gpk_ctx.ui_approve_l2, "Approve", "Reject", callback_export);
```

Comment:
Why not using nbgl_useCaseReviewStart() as done in various porting like https://github.com/LedgerHQ/app-boilerplate/blob/master/src/ui/nbgl_display_address.c#L100

Reply:
bad copy paste from a modified-since app

Reply:
Done

Reply:
The Hedera app has a flow where the user can allow or refuse to export the public key, I don't really know why but as it's an accepted behavior according to the guidelines I've let it be this way for Stax too

Reply:
If that the wanted UX, lets keep it!